src/realm/sync/network/http.hpp

```diff
@@ -306,6 +309,15 @@ struct HTTPParser : protected HTTPParserBase {
         m_socket.async_read_until(m_read_buffer.get(), max_header_line_length, '\n', std::move(handler));
     }
 
+    inline uint32_t hex_to_int(const std::string& hex) throw()
+    {
+        uint32_t dec;
+        std::stringstream ss;
+        ss << std::hex << hex;
+        ss >> dec;
```

Comment:
Does this do any kind of bounds checking in the input? What happens if the number in `hex` is larger than a `uint32_t`?